packages/frontend/src/hooks/useCartesianChartConfig.ts

```diff
@@ -210,12 +212,18 @@ const useCartesianChartConfig = (
                 yField: validYFields.length > 0 ? validYFields : yField,
             };
         };
+
+        const setPivotField = (pivotField: string) => {
+            setPivotDimensions((currentPivots) => {
+                return currentPivots === undefined
+                    ? [pivotField]
+                    : currentPivots;
+            });
+        };
+
         // Only load this if there are no existing chart configuration (not saved)
         if (isSaved) return;
 
-        setPivotDimensions([]); //reset pivot
-        setVisualizationChartType(ChartType.CARTESIAN); // reset table
```

Comment:
@rephus what was the intention with these two lines?

Reply:
The intention with those 2 lines is to reset the visualization chart and the pivot dimensions. For example, if you have `two metrics, two dimensions` it shows 1 pivot, if now you deselect 1 dimension, it should remove the pivot (by doing `setPivotDimensions([]);` or `setPivotDimensions(undefined)` ) Otherwise, the pivot will stay in place. If you want to remove those lines, you'll have to set the right pivotDimension and visualizationCharrType inside every `IF`

Reply:
Ah yes makes sense. For `pivotDimensions` I copied your pattern for `setAxes` by using existing values if they've been set.

A problem with the current approach, is that the default write the same state that the user does so it's impossible to tell when the state is user-defined vs default-defined. So when I was changing the pivot dimension, this code was removing my preferences when hitting "run query". e.g. if I change to a big number viz, then hit run query, it would be changed back to a bar chart!

I'll try a similar approach with chart, where we can auto-set it if it's undefined otherwise leave it.

I don't think it's perfect but after this fix we could figure out another way to apply defaults that don't manipulate the user-defined state